modules/layers/src/icon-layer/icon-manager.js

```diff
@@ -307,8 +310,15 @@ export default class IconManager {
   _loadIcons(icons) {
     const ctx = this._canvas.getContext('2d');
 
+    let loaded = 0;
     for (const icon of icons) {
       loadImage(icon.url).then(imageData => {
+        loaded++;
```

Comment:
More scenarios to consider:

- `loaded` will never be set if one of the images fails to load, e.g. url is not found. Is that intended?
- This logic breaks if data is changed and a new batch of icons start loading. How do you handle that?